python/understack-workflows/understack_workflows/main/undersync_device.py

```diff
@@ -23,20 +22,73 @@ def update_nautobot(args) -> UUID:
     nb_url = args.nautobot_url
     nb_token = args.nautobot_token or credential("nb-token", "token")
 
-    new_status = network_name_status[args.network_name]
+    logger.info(f"Updating Nautobot {device_id=!s} {interface_mac=!s} {network_name=}")
+
+    if network_name == "tenant":
+        switch_id = update_nautobot_for_tenant(
+            nb_url, nb_token, interface_mac, args.network_id
+        )
+    elif network_name == "provisioning":
+        switch_id = update_nautobot_for_provisioning(
+            nb_url, nb_token, device_id, interface_mac
+        )
+    else:
+        raise ValueError(f"need provisioning or tenant, not {network_name=}")
+
+    logger.info(f"Updated Nautobot {device_id=!s} {interface_mac=!s} {network_name=}")
+
+    logger.info(f"Interface connected to switch {switch_id!s}")
+    return switch_id
+
 
+def update_nautobot_for_provisioning(
+    nb_url, nb_token, device_id: UUID, interface_mac: str
+):
+    new_status = "Provisioning Interface"
     nautobot = Nautobot(nb_url, nb_token, logger=logger)
-    logger.info(f"Updating Nautobot {device_id=!s} {interface_mac=!s} {network_name=}")
+
     interface = nautobot.update_switch_interface_status(
         device_id, interface_mac, new_status
     )
-    logger.info(f"Updated Nautobot {device_id=!s} {interface_mac=!s} {network_name=}")
-
     switch_id = interface.device.id
-    logger.info(f"Interface connected to switch {switch_id!s}")
     return switch_id
 
 
+def update_nautobot_for_tenant(
+    nb_url, nb_token, server_interface_mac: str, ucvni_id: UUID
+) -> UUID:
+    """Runs a Nautobot Job to update a switch interface for tenant mode
+
+    The nautobot job will assign vlans as required and set the interface
+    into the correct mode for "normal" tenant operation.
+
+    The switch ID is returned.
+    """
+
+    # Making this http request directly because it was not clear how to get
+    # the pynautobot api client to call an arbitrary endpoint:
+
+    uri = f"{nb_url}api/plugins/undercloud-vni/ucvni_jobs/prep_switch_interface"
```

Comment:
The default nb url does NOT contain ending "/":
https://github.com/rackerlabs/understack/blob/f024a0e7ae9b0e39b081f6878cbe43aa045b74aa/python/understack-workflows/understack_workflows/helpers.py#L44

Reply:
good catch :)